Python/pylifecycle.c

```diff
@@ -1272,30 +1274,30 @@ init_interp_main(PyThreadState *tstate)
     }
 
     // Turn on experimental tier 2 (uops-based) optimizer
+    // This is also needed when the JIT is enabled
+#ifdef _Py_TIER2
     if (is_main_interp) {
```

Comment:
I know this briefly came up in the meeting, but why are we only doing this in the main interpreter again? (Doesn't necessarily need to be fixed here, just remembered and hot curious.)

Reply:
I have no idea any more. I'll see if deleting this check causes any problems.

Reply:
Alas, deleting it does cause problems -- `test.test_capi.test_misc.TestPendingCalls.test_isolated_subinterpreter` hangs. So I'm gonna stick with what works.